### ISSUE TITLE
Problem handling certain variable names?

### ISSUE THREAD

**Issue by jmbejara** [bashtage/linearmodels#149]
Hi. I'm not sure what's causing this issue, but I thought I might bring it up here in case it's useful. Thanks for the great package!!

When I run the following, I get an error:

```python
import numpy as np
np.set_printoptions(suppress=True, precision=2)
import pandas as pd
import linearmodels

def generate_data(N=50):
    mean = np.array([0,0,0])
    cov = np.array([[1, 0, .5],
                    [0, 1, 0],
                    [.5, 0, 1]])
    vec = np.random.multivariate_normal(mean, cov, N)
    df = pd.DataFrame(vec, columns=['u', 'z', 'epsilon'])
    beta0 = beta1 = gamma0 = gamma1 = 1
    beta2 = 2
    df['x1'] = gamma0 + gamma1 * df.z + df.u
    df['y'] = beta0 + beta1 * df.x1 + df.epsilon
    return df

np.random.seed(100)
df = generate_data(N=100)
reg = linearmodels.IV2SLS.from_formula('y ~ 1 + [x1 ~ z]', data=df).fit()
```

ERROR:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-1-e7558e367513> in <module>()
     19 np.random.seed(100)
     20 df = generate_data(N=100)
---> 21 reg = linearmodels.IV2SLS.from_formula('y ~ 1 + [x1 ~ z]', data=df).fit()

~\Anaconda3\lib\site-packages\linearmodels\iv\model.py in from_formula(formula, data, weights)
    618         >>> mod = IV2SLS.from_formula(formula, data)
    619         """
--> 620         parser = IVFormulaParser(formula, data)
    621         dep, exog, endog, instr = parser.data
    622         mod = IV2SLS(dep, exog, endog, instr, weights=weights)

~\Anaconda3\lib\site-packages\linearmodels\iv\_utility.py in __init__(self, formula, data, eval_env)
     78         self._eval_env = eval_env
     79         self._components = {}
---> 80         self._parse()
     81 
     82     def _parse(self):

~\Anaconda3\lib\site-packages\linearmodels\iv\_utility.py in _parse(self)
     99                 raise ValueError('endogenous block must not start or end with +. This block '
    100                                  'was: {0}'.format(endog))
--> 101             if instr[0] == '+' or instr[1] == '+':
    102                 raise ValueError('instrument block must not start or end with +. This '
    103                                  'block was: {0}'.format(instr))

IndexError: string index out of range
```

If I change the variable name, then the error goes away:
```python
df['z1'] = df.z
reg = linearmodels.IV2SLS.from_formula('y ~ 1 + [x1 ~ z1]', data=df).fit()
reg
```

I'm using Python 3.6 on Anaconda. Here is the version info of linearmodels and pandas:
```
λ pip show linearmodels
Version: 4.7
```
```
λ pip show pandas
Version: 0.22.0
```

**Comment by bashtage** [comment]
Thanks. Pretty sure this is a bug that arises when the instrument block has
a single character. Should be an easy fix. Should be instr[-1] instead of
instr[1]

**Comment by bashtage** [comment]
Thanks - this has been fixed in 4.8, now on PyPi.

**Comment by jmbejara** [comment]
OK. That was fast! Thanks again for this package!
